pkg/driver/node.go

```diff
@@ -71,30 +73,56 @@ func (d *Driver) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRe
 		return nil, status.Error(codes.InvalidArgument, "Volume capability not supported")
 	}
 
+	if ok := d.inFlight.Insert(req); !ok {
```

Comment:
In this scenario, I believe we do want the `NodeUnstageVolume` request to fail and keep retrying.

For example, `NodeStageVolume` is called when a new pod is being created, the user then deletes that pod and PVC before its ready and the volume was staged, this then triggers the `NodeUnstageVolume`, if `target` is used instead of `req` and no error is returned the controller won't retry and the volume will never actually get unstaged.

Reply:
Create a separate issue to track unstage idempotency #175